wandb/sdk/service/streams.py

```diff
@@ -247,41 +242,51 @@ def _process_drop(self, action: StreamAction) -> None:
                 stream.drop()
                 stream.join()
 
-    def _on_probe_exit(self, probe_handle: MailboxProbe, stream: StreamRecord) -> None:
-        handle = probe_handle.get_mailbox_handle()
-        if handle:
-            result = handle.wait(timeout=0, release=False)
-            if not result:
-                return
-            probe_handle.set_probe_result(result)
-        handle = stream.interface.deliver_poll_exit()
-        probe_handle.set_mailbox_handle(handle)
-
-    def _on_progress_exit(self, progress_handle: MailboxProgress) -> None:
-        pass
-
-    def _on_progress_exit_all(
+    async def _finish_all_progress(
```

Comment:
would be good to fix this behavior for wandb-core eventually, currently we don't handle implicit finish very well :(

Reply:
yeah absolutely, something I'd be happy to work on later.

Reply:
yep, i think we should slate it... i will take a note on that